typescript/packages/wallets/crossmint/src/mintingAPI.ts

```diff
@@ -0,0 +1,128 @@
+import { z } from "zod";
+import type { Plugin, WalletClient } from "@goat-sdk/core";
+import { randomUUID } from "node:crypto";
+
+export const mintingAPIFactory = (
+    apiKey: string,
+): ((options: { env: "production" | "staging" }) => Plugin<WalletClient>) => {
```

Comment:
i think u should just install `@crossmint/common-sdk-base`, and use `validateAPIKey` which will return the environment for the api key

then use `environmentToCrossmintBaseURL`

Reply:
so its not necessary to pass `env`, and its impossible to have this out of sync with the env the api key is actually for

Reply:
I agree there are a lot of better ways to do this but I'm just following the pattern that already exists for plugins for the purposes of this PR

Reply:
theres a single other place that does it like that, and its not a breaking change to do it here, so i dont see any reason why we would not do it the correct way

Reply:
Added this